replication.go

```diff
@@ -69,44 +113,63 @@ func (r *ReplicationState) collectMissingSequences(observedSeq uint64, nextSeqTo
 }
 
 // sendReplicationRequests sends requests for missing sequences for the
-// range of sequences [start, end] <- inclusive
+// range of sequences [start, end] <- inclusive. It does so by splitting the
+// range of sequences equally amount the nodes that have signed [highestSequenceObserved].
 func (r *ReplicationState) sendReplicationRequests(start uint64, end uint64) {
+	nodes := r.highestSequenceObserved.getSigners()
+
+	// round up to ensure we get all the sequences
+	reqPerNode := uint64(math.Ceil(float64(end+1-start) / float64(len(nodes))))
+
+	nodeIndex := r.requestIterator
+
+	// <= because we are inclusive
+	for curSeq := uint64(0); curSeq <= end+1-start; curSeq += reqPerNode {
+		endSeq := uint64(math.Min(float64(end), float64(start+curSeq+reqPerNode)))
+		index := nodeIndex % len(nodes)
+		// it's possible our node has signed [highestSequenceObserved].
+		// For example this may happen if our node has sent a finalization
```

Comment:
> // in this case we shouldn't send a request to ourselves.

We shouldn't send the request to ourselves in other cases as well. 
I think for simplicity instead of doing the `if` below we can just filter out our own ID above when we get the nodes to replicate from.